backend/allocation/graphql/query/courseAllocation.py

```diff
@@ -0,0 +1,56 @@
+
+import graphene
+from allocation.graphql.types.allocation import CourseAndFacultyType,AllocationFilterInput
+from course.graphql.types.course import CourseType
+from allocation.models import CourseAllocation
+from user.models import Faculty
+from user.graphql.types.user import FacultyType
+from backend.api import APIException
+from backend.api.decorator import login_required
+
+class AllocationQueries(graphene.ObjectType):
+    all_allocation = graphene.List(CourseAndFacultyType,)
+    # batch_wise_allocation = graphene.List(CourseAndFacultyType
+    allocation = graphene.List(
+        CourseAndFacultyType,
+        filter = graphene.Argument(AllocationFilterInput,required=True))
+    
+    @login_required
+    def resolve_allocation(self,info,filter):
+        if filter :
+            if filter.batch and filter.year and filter.sem :
+                allocations = CourseAllocation.objects.all()
+                allocations = allocations.filter(
+                    course__batch__curriculum__program__name=filter.batch,
+                    course__batch__year = filter.year,
+                    course__batch__sem = filter.sem,
+                )
+                course_faculty_list = []
+                for allocation in allocations:
+                    course = allocation.course
+                    user = allocation.faculty
+                    try:
+                        faculty = Faculty.objects.get(id=user.id)
```

Comment:
instead of querying for faculty inside every allocation iteration, figure out a way to do this in 1 or 2 query.